models/ios/snmp_server/ios_snmp_server.yaml

```diff
@@ -0,0 +1,872 @@
+                type: bool
+              linkdown:
+                description: Enable linkDown trap
+                type: bool
+              linkup:
+                description: Enable linkUp trap
+                type: bool
+              warmstart:
+                description: Enable warmStart trap
+                type: bool
+            type: dict
+          syslog:
+            description: Enable SNMP syslog traps
+            type: bool
+          transceiver_all:
+            description: Enable SNMP transceiver traps
+            type: bool
+          tty:
+            description: Enable SNMP tty TCP connection traps
+            type: bool
+          vrrp:
+            description: Enable SNMP vrrp traps
+            type: bool
+        type: dict
+      user:
+        description: Define a user who can access the SNMP engine
+        elements: dict
+        suboptions:
+          access:
+            description: Access list associated
+            type: str
+          group:
+            description: SNMP group for the user.
+            type: str
+          remote:
+            description: System where an SNMPv3 user is hosted
+            type: str
+          udp_port:
+            description: UDP port used by the remote SNMP system
+            type: int
+          user:
+            description: SNMP user name
+            type: str
+          version:
+            choices:
+              - v1
+              - v2c
+              - v3
+            description: SNMP security version
+            type: str
+          vrf:
+            description: The remote SNMP entity's VPN Routing instance
+            type: str
+        type: list
+      view:
+        description: Define an SNMPv2 MIB view
+        elements: dict
+        suboptions:
+          excluded:
+            description: MIB family is excluded from the view
+            type: bool
+          family_name:
+            description: MIB view family name
+            type: str
+          included:
+            description: MIB family is included in the view
+            type: bool
+          name:
+            description: Name of the view
+            type: str
+        type: list
+    type: dict
+  running_config:
+    description:
+      - This option is used only with state I(parsed).
+      - The value of this option should be the output received from the IOS device by executing the command B(show running-config | include snmp-server).
+      - The state I(parsed) reads the configuration from C(running_config) option and transforms it into Ansible structured data as per the resource module's argspec and the value is then returned in the I(parsed) key within the result.
+    type: str
+  state:
+    choices:
+      - merged
+      - replaced
+      - overridden
+      - deleted
+      - parsed
+      - gathered
+      - rendered
+    default: merged
+    description:
+      - The state the configuration should be left in.
```

Comment:
- The states I(replaced) and I(overridden) have identical
       behaviour for this module.